core/src/main/java/com/linecorp/armeria/server/annotation/BeanRequestConverterFunction.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.validateAndNormalizeSupportedType;
+import static org.reflections.ReflectionUtils.getAllFields;
+import static org.reflections.ReflectionUtils.getAllMethods;
+import static org.reflections.ReflectionUtils.withParametersCount;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+
+import com.google.common.collect.MapMaker;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+/**
+ * A default {@link RequestConverterFunction} which converts a request into a Java bean.
+ *
+ * <p>The following parameters are eligible for automatic injection into bean fields or properties:
+ * <ul>
+ *   <li>Path parameters to fields annotated by {@link Param}</li>
+ *   <li>HTTP parameters to fields annotated by {@link Param}</li>
+ *   <li>Headers to fields annotated by {@link Header}</li>
+ * </ul>
+ */
+public class BeanRequestConverterFunction implements RequestConverterFunction {
+    private static final ConcurrentMap<Class<?>, BeanParamInfo> beanParamInfoCache = new MapMaker().weakKeys()
+                                                                                                   .makeMap();
+
+    private static final ConcurrentMap<Class<?>, Boolean> ignoredResultTypeCache = new MapMaker().weakKeys()
+                                                                                                 .makeMap();
+
+    /**
+     * Builds and registers the parameter information for {@code expectedResultType}.
+     *
+     * @param expectedResultType a candidate data type, which will be converted by ${@link BeanRequestConverterFunction}
```

Comment:
The length of this line exceeds the limit which is 112.
Don't think we need `$`

Reply:
Fixed the line length problem.

About the `${@link BeanRequestConverterFunction}`, please see this comment: https://github.com/line/armeria/pull/1096#discussion_r176300804